Scripts/RemediationScripts/Remediate-EnableHTTPSForAppServices.ps1

```diff
@@ -0,0 +1,917 @@
+        
+        $validAppServiceDetails | ForEach-Object {
+            $resourceId = $_.ResourceId
+
+            try
+            {
+                Write-Host "Fetching App Service resource: Resource ID - $($resourceId)"
+                $appServiceResource = Get-AzResource -ResourceId $resourceId -ErrorAction SilentlyContinue
+                $appServiceResources += $appServiceResource
+            }
+            catch
+            {
+                Write-Host "Error fetching App Service resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping this App Service resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    }
+
+    $totalAppServices = ($appServiceResources | Measure-Object).Count
+
+    if ($totalAppServices -eq 0)
+    {
+        Write-Host "No App Services found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+  
+    Write-Host "Found $($totalAppServices) App Service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Includes App Services where HTTPS is enabled on all slots - production slot and all non-production slots.
+    $appServicesWithHttpsEnabled = @()
+
+    # Includes App Services where HTTPS is not enabled on all slots - production slot or one or more non-production slots.
+    $appServicesWithoutHttpsEnabled = @()
+
+    # Includes App Services that were skipped during remediation. There were errors remediating them.
+    $appServicesSkipped = @()
+
+    $appServiceResources | ForEach-Object {
+        $appServiceResource = $_
+        $resourceId = $_.ResourceId
+        $resourceGroupName = $_.ResourceGroupName
+        $resourceName = $_.ResourceName
+
+        try
+        {
+            Write-Host "Fetching App Service configuration: Resource ID - $($resourceId), Resource Group Name - $($resourceGroupName), Resource Name - $($resourceName)"
+            
+            $isHttpsEnabledOnProductionSlot = $(Get-AzWebApp -ResourceGroupName $resourceGroupName -Name $appServiceResource.Name -ErrorAction SilentlyContinue).HttpsOnly
+
+            Write-Host "Fetching non-production slot configurations for App Service: Resource ID - $($resourceId)"
+
+            # Get all non-production slots for this App Service.
+            $nonProductionSlotConfigurations = Get-AzWebAppSlot -ResourceGroupName $resourceGroupName -Name $resourceName
+
+            $isHttpsEnabledOnAllNonProductionSlots = -not $($nonProductionSlotConfigurations.HttpsOnly -contains $false)
+
+            if ($isHttpsEnabledOnProductionSlot -and $isHttpsEnabledOnAllNonProductionSlots)
+            {
+                $appServicesWithHttpsEnabled += $appServiceResource
+                Write-Host "HTTPS is enabled on the production slot and all non-production slots in the App Service." -ForegroundColor $([Constants]::MessageType.Update)
+                return
```

Comment:
Plz check will return just skip the current iteration of loop or returns from loop entirely.

Reply:
`return` will skip the current iteration of a `ForEach-Object` block. Reference: https://stackoverflow.com/questions/7760013/why-does-continue-behave-like-break-in-a-foreach-object